### PR TITLE
[js/webgpu] Add uniforms support to concat op

### PR DESCRIPTION
**Pull request by axinging**

### CODE REVIEW

**Comment by axinging** [comment]
@qjia7 @xhcao @hujiajie @gyagp PTAL

File: js/web/lib/wasm/jsep/webgpu/ops/concat.ts [hunk 1]
```diff
@@ -92,40 +92,62 @@ const createConcatProgramInfo = (inputs: readonly TensorView[], axis: number): P
...
+  const inputShapeOrRanks = [];
+  const enableInputShapesUniforms = [];
+  let programUniforms: ProgramUniform[] = [{type: 'uint32', data: outputSize}];
   for (let i = 0; i < inputs.length; ++i) {
     previousSum += inputs[i].dims[adjustedAxis];
     sizeInConcatAxis[i] = previousSum;
+    enableInputShapesUniforms.push(enableShapesUniforms(inputs[i].dims.length));
+    inputShapeOrRanks.push(enableInputShapesUniforms[i] ? inputs[i].dims.length : inputs[i].dims);
+    inputVars[i] = inputVariable(`input${i}`, dataType, inputShapeOrRanks[i]);
+    inputDependencies.push('rank');
```
Comment (qjia7) [inline js/web/lib/wasm/jsep/webgpu/ops/concat.ts:105]: inputDependencies.push(enableInputShapesUniforms[i] ? 'rank' : 'dims');?

Comment (axinging) [reply js/web/lib/wasm/jsep/webgpu/ops/concat.ts]: Done， thanks!

**Review by qjia7 — APPROVED** [review]
The code looks good to me.
Please try t5-decoder to make sure your changes do work.
@guschmue @satyajandhyala  Please take a look, thanks.

File: js/web/lib/wasm/jsep/webgpu/ops/concat.ts [hunk 2]
```diff
@@ -92,40 +92,62 @@ const createConcatProgramInfo = (inputs: readonly TensorView[], axis: number): P
...
-  const output = outputVariable('output', dataType, outputShape);
+  const outputShapeOrRank = enableOutputShapesUniforms ? outputShape.length : outputShape;
+  const output = outputVariable('output', dataType, outputShapeOrRank);
 
   const indicesAxis = output.indicesGet('indices', adjustedAxis);
   const getShaderSource = (shaderHelper: ShaderHelper) => `
-  ${shaderHelper.declareVariables(...inputVars, output)}
+  ${
+      shaderHelper.registerUniform('outputSize', 'u32')
+          .registerUniform(`sizeInConcatAxis`, `vec${inputs.length}<u32>`)
```
Comment (qjia7) [inline js/web/lib/wasm/jsep/webgpu/ops/concat.ts:126]: Sorry, one more question: Is that possible that inputs.length > 4? If yes, there is potential bug here.

**Comment by guschmue** [comment]
lint nagging:
image

**Comment by axinging** [comment]
> lint nagging: image

Done.

**Review by guschmue — APPROVED** [review]

**Pull request merged**